system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj

```diff
@@ -27,64 +27,103 @@
+     (is (= 1 (count (:entry (:feed data)))))))
 
   (testing "bad value search with no results"
-   (let [response (search/get-autocomplete-suggestions "zee")
+   (let [response (search/get-autocomplete-suggestions "q=zee")
          body (:body response)
          data (json/parse-string body true)]
-     (is (= 0 (count (:items (:results data)))))))
+     (is (= 0 (count (:entry (:feed data)))))))
 
   (testing "search with type filter"
-   (let [response (search/get-autocomplete-suggestions "foo" ["instrument"] nil)
+   (let [response (search/get-autocomplete-suggestions "q=foo&type[]=instrument")
          body (:body response)
          data (json/parse-string body true)]
-     (is (= 1 (count (:items (:results data)))))))
+     (is (= 1 (count (:entry (:feed data)))))))
 
   (testing "search with multiple types filter"
-   (let [response (search/get-autocomplete-suggestions "foo" ["instrument" "platform" "project" "provider"] nil)
+   (let [response (search/get-autocomplete-suggestions
+                    "q=b&type[]=instrument&type[]=platform&type[]=project&type[]=provider")
          body (:body response)
          data (json/parse-string body true)]
-     (is (= 1 (count (:items (:results data)))))))
+     (is (= 2 (count (:entry (:feed data)))))))
 
   (testing "search with type filter with no results"
-   (let [response (search/get-autocomplete-suggestions "foo" ["platform"] nil)
+   (let [response (search/get-autocomplete-suggestions "q=foo&type[]=platform")
          body (:body response)
          data (json/parse-string body true)]
-     (is (= 0 (count (:items (:results data))))))))
+     (is (= 0 (count (:entry (:feed data))))))))
 
 (deftest autocomplete-usage-test
   (testing "missing query param response test"
-           (let [response (search/get-autocomplete-suggestions nil {:throw-exceptions false})]
-             (is (= 400 (:status response))))))
+   (let [response (search/get-autocomplete-suggestions nil {:throw-exceptions false})]
+     (is (= 400 (:status response)))))
+
+  (testing "empty string query response test"
+   (let [response (search/get-autocomplete-suggestions "q=  " {:throw-exceptions false})]
+     (is (= 400 (:status response)))))
+
+  (testing "page_size 0 test"
+   (let [response (search/get-autocomplete-suggestions "q=*&page_size=0")
+         headers (:headers response)
+         body (json/parse-string (:body response) true)
+         entry (:entry (:feed body))]
+     (is (= "3" (:CMR-Hits headers)))
+     (is (= 0 (count entry)))))
+
+  (testing "page_size 2 test"
+   (let [response (search/get-autocomplete-suggestions "q=*&page_size=2")
+         headers (:headers response)
+         body (json/parse-string (:body response) true)
+         entry (:entry (:feed body))]
+     (is (= "3" (:CMR-Hits headers)))
+     (is (= 2 (count entry)))))
+
+  (testing "page_num should yield different 'first' results test"
+   (let [page-one-response (search/get-autocomplete-suggestions "q=b&page_size=1")
```

Comment:
This is a good place to use the thread first macro `->` instead of creating a bunch of bindings that are only used to create the next binding.

Reply:
I think this one is a little too complicated for just thread-first and would need something like `as->`